services/tenant-ui/frontend/src/components/contacts/messageContact/MessageContactList.vue

```diff
@@ -0,0 +1,286 @@
+<template>
+  <div class="title">
+    <h2>Chat with {{ props.connectionName }}</h2>
+  </div>
+  <div class="container">
+    <div
+      v-for="item in messageList"
+      :key="item.message_id"
+      :class="item.state === 'received' ? 'theirs' : 'mine'"
+      class="message"
+    >
+      <div class="bubble">
+        {{ item.content }}
+      </div>
+      <div class="time" :class="{ display: item.displayTime }">
+        {{ formatTime(item.sent_time) }}
+      </div>
+    </div>
+  </div>
+</template>
+<script setup lang="ts">
+// Vue
+import { ref, PropType, watch, onUpdated, onMounted } from 'vue';
+import { storeToRefs } from 'pinia';
+
+// State
+import { useMessageStore } from '@/store';
+
+/**
+ * Display the time or not.
+ * If the time is more than 24 hours ago then only
+ * display the date of the last message.
+ * @param time string
+ * @param who string
+ * @return boolean
+ */
+// const displayTime = (time: string, who: string) => {
+//   const date: any = new Date(time);
+//   const now: any = new Date();
+//   if (now - date > 86400000) {
+//     return true;
+//   } else {
+//     return false;
+//   }
+// }
+
+/**
+ * formatTime
+ * Format the time to be displayed in the chat window.
+ * @param time string
+ */
+const formatTime = (time: string) => {
+  const date: any = new Date(time);
+  const now: any = new Date();
+  if (now - date > 86400000) {
+    const options = {
+      weekday: 'long',
+      year: 'numeric',
+      month: 'long',
+      day: 'numeric',
+    };
+    return date.toLocaleDateString('en-CA', options);
+  } else {
+    return date.toLocaleTimeString();
+  }
+};
```

Comment:
Agreed. I think it's best to use the formatters that are default to all browsers. https://caniuse.com/date-tolocaledatestring

Reply:
Don't worry too much about legacy browser support because Vite uses ES Modules https://caniuse.com/es6-module so anything too old is already thrown out and we don't support.